# Global HLA class I frequencies and predicted binding affinities to HIV peptides

The visualizations and data showcased in this app are an effort to present and summarize the interaction between the Human Leucocyte Antigen (HLA) class I alleles and HIV viral subtypes.

We provide our results on imputed HLA alleles and their predicted binding affinity to HIV peptides as represented in the global START cohort from the [INSIGHT Consortium](http://insight.ccbr.umn.edu/).

## Reference
If you use any of the results or data in this web application please cite the corresponding article:

Zucco, A. G., Bennedbæk, M., Ekenberg, C., Gabrielaite, M., Leung, P., Polizzotto, M. N., Kan, V., Murray, D. D., Lundgren, J. D., & MacPherson, C. R. (2023). Associations of functional human leucocyte antigen class I groups with HIV viral load in a heterogeneous cohort. AIDS, 37(11), 1643. https://doi.org/10.1097/QAD.0000000000003557

**Abstract:**

> **Objective**: <br> Human leucocyte antigen (HLA) class I alleles are the main host genetic factors involved in controlling HIV-1 viral load (VL). Nevertheless, HLA diversity has proven a significant challenge in association studies. We assessed how accounting for binding affinities of HLA class I alleles to HIV-1 peptides facilitate association testing of HLA with HIV-1 VL in a heterogeneous cohort. <br> <br>**Design**: <br> Cohort from the Strategic Timing of AntiRetroviral Treatment (START) study. <br> <br>**Methods**: <br> We imputed HLA class I alleles from host genetic data (2546 HIV+ participants) and sampled immunopeptidomes from 2079 host-paired viral genomes (targeted amplicon sequencing). We predicted HLA class I binding affinities to HIV-1 and unspecific peptides, grouping alleles into functional clusters through consensus clustering. These functional HLA class I clusters were used to test associations with HIV VL.<br> <br>**Results**: <br> We identified four clades totaling 30 HLA alleles accounting for 11.4% variability in VL. We highlight HLA-B∗57:01 and B∗57:03 as functionally similar but yet overrepresented in distinct ethnic groups, showing when combined a protective association with HIV+ VL (log, β −0.25; adj. P-value < 0.05). We further demonstrate only a slight power reduction when using unspecific immunopeptidomes, facilitating the use of the inferred functional HLA groups in other studies. <br> <br>**Conclusion**: <br> The outlined computational approach provides a robust and efficient way to incorporate HLA function and peptide diversity, aiding clinical association studies in heterogeneous cohorts. To facilitate access to the proposed methods and results we provide an interactive application for exploring data.

## Data Sources

To facilitate open-access to our results and non-sensitive data please visit [Downloads].
The raw data of this project correspond to patients living with HIV hence the need for confidentiality. Access to the [START trial](http://insight.ccbr.umn.edu/start/) data can be requested to the [INSIGHT Consortium](http://insight.ccbr.umn.edu/). This project has been conducted as a collaboration between INSIGHT and the Centre of Excellence for Personalised Medicine of Infectious complications in immune deficiency ([PERSIMUNE](http://www.persimune.dk/)).

## License

This work is completely free and open for academic use and furthering research in human immunology and infectious disease; provided full attribution is given.

## Acknowledgments

We would like to thank all participants in the START trial and all trial investigators. See [N Engl J Med 2015; 373:795–807(20)](https://www.nejm.org/doi/full/10.1056/nejmoa1506816) for the complete list of START investigators. 

Many thanks to the authors and maintainers of all of the R packages used to make this app. A great thanks goes to Sean Angiolillo who wrote the original code and from whom this project was forked (see the original blog post [here](https://blog.socialcops.com/technology/data-science/comparative-thematic-mapping/)).
